src/modules/ekf2/EKF/aid_sources/external_vision/ev_vel.h

```diff
@@ -0,0 +1,176 @@
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file ev_vel.h
+ * Handling different types of External Vision velocity measurements
+ */
+
+#ifndef ExternalVisionVel_H
+#define ExternalVisionVel_H
+
+#include "ekf.h"
+#include <uORB/topics/estimator_aid_source3d.h>
+
+class ExternalVisionVel
+{
+public:
+	ExternalVisionVel(Ekf &ekf_instance, const extVisionSample &vision_sample, const float ev_vel_noise,
+			  const imuSample &imu_sample) : _ekf(ekf_instance), _sample(vision_sample)
+	{
+		_min_variance = sq(ev_vel_noise);
+		const Vector3f angular_velocity = imu_sample.delta_ang / imu_sample.delta_ang_dt - _ekf._state.gyro_bias;
+		Vector3f _position_offset_body = _ekf._params.ev_pos_body - _ekf._params.imu_pos_body;
+		_velocity_offset_body = angular_velocity % _position_offset_body;
+	}
+
+	virtual ~ExternalVisionVel() = default;
+	virtual bool fuseVelocity(estimator_aid_source3d_s &aid_src, float gate)
+	{
+		_ekf.fuseLocalFrameVelocity(aid_src, aid_src.timestamp, _measurement,
+					    _measurement_var, gate);
+		return aid_src.fused;
+
+	}
+
+	virtual void resetVelocity()
+	{
+		_ekf.resetVelocityTo(_measurement, _measurement_var);
+	}
+
+	void resetVerticalVelocity()
+	{
+		_ekf.resetVerticalVelocityTo(_measurement(2), _measurement_var(2));
+	}
+
+	void enforceMinimumVariance()
+	{
+		_min_variance = math::max(_min_variance, sq(0.01f));
+
+		for (int i = 0; i < 3; ++i) {
+			_measurement_var(i) = math::max(_measurement_var(i), _min_variance);
+		}
+	}
+
+	Ekf &_ekf;
+	const extVisionSample &_sample;
+	float _min_variance;
+	Vector3f _measurement{NAN, NAN, NAN};
+	Vector3f _measurement_var;
+	Vector3f _velocity_offset_body;
+};
+
+class EvVelBodyFrameFrd : public ExternalVisionVel
+{
+public:
+	EvVelBodyFrameFrd(Ekf &ekf_instance, extVisionSample &vision_sample, const float ev_vel_noise,
+			  const imuSample &imu_sample) :
+		ExternalVisionVel(ekf_instance, vision_sample, ev_vel_noise, imu_sample)
+	{
+		_measurement = _sample.vel - _velocity_offset_body;
+		_measurement_var = _sample.velocity_var;
+		enforceMinimumVariance();
+	}
+
+	void resetVelocity() override
+	{
+		const matrix::SquareMatrix3f rotated_variance = _ekf._R_to_earth * matrix::diag(
+					_measurement_var) * _ekf._R_to_earth.transpose();
+		// bump the variance by a factor of 5 to make teh reset less aggressive
```

Comment:
please explain why we want to make the reset less aggressive here. Also remove the "by a factor of 5" since this is obvious by reading the code